docker/docker.go

```diff
@@ -101,6 +105,17 @@ func New() (*docker, error) {
 func (d *docker) CollectMetrics(mts []plugin.Metric) ([]plugin.Metric, error) {
 	var err error
 	metrics := []plugin.Metric{}
+
+	conf, err := getDockerConfig(mts[0])
+	if err != nil {
+		return nil, err
+	}
+
+	err = d.init_client(conf["endpoint"])
```

Comment:
@marcin-krolik take a look at new commit, in `initClient` method, we check if client was initialized, if yes (checking if `.client` is not `nil`, but i think it would be more useful to do some sort of `ping` check if Docker daemon is alive and its API is responsive).